### ISSUE TITLE
Type mismatch with dns_type question

### ISSUE THREAD

**Issue by ColonelPanicks** [alces-software/metalware#179]
PR with question for metalware-default: alces-software/metalware-default#11

After running a configure on the domain it is not possible to run the command again:
```
[root@controller repo]# metal configure domain
error: Failed to validate answers: /var/lib/metalware/answers/domain.yaml
A type mismatch has been detected in the following question(s):
{:question=>:dns_type, :answer=>"named", :type=>"choice"}
. Use --trace to view backtrace
[root@controller repo]# metal configure domain --trace
/opt/metalware/src/validation/answer.rb:78:in `load': Failed to validate answers: /var/lib/metalware/answers/domain.yaml (Metalware::ValidationFailure)
A type mismatch has been detected in the following question(s):
{:question=>:dns_type, :answer=>"named", :type=>"choice"}
	from /opt/metalware/src/validation/loader.rb:66:in `answer'
	from /opt/metalware/src/validation/loader.rb:47:in `domain_answers'
	from /opt/metalware/src/dependency.rb:76:in `validate_answer_file'
	from /opt/metalware/src/dependency.rb:69:in `validate_dependency_value'
	from /opt/metalware/src/dependency.rb:54:in `block (2 levels) in run_dependencies'
	from /opt/metalware/src/dependency.rb:54:in `each'
	from /opt/metalware/src/dependency.rb:54:in `block in run_dependencies'
	from /opt/metalware/src/dependency.rb:48:in `each'
	from /opt/metalware/src/dependency.rb:48:in `run_dependencies'
	from /opt/metalware/src/dependency.rb:40:in `enforce'
	from /opt/metalware/src/command_helpers/base_command.rb:79:in `enforce_dependency'
	from /opt/metalware/src/command_helpers/base_command.rb:59:in `post_setup'
	from /opt/metalware/src/command_helpers/base_command.rb:39:in `initialize'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/command.rb:217:in `new'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/command.rb:217:in `call'
	from /opt/metalware/src/commander_extensions.rb:61:in `call'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/command.rb:184:in `run'
	from /opt/metalware/src/commander_extensions.rb:49:in `run'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/runner.rb:464:in `run_active_command'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/runner.rb:66:in `run!'
	from /opt/metalware/vendor/ruby/2.4.0/bundler/gems/commander-c99203ec5a74/lib/commander/delegates.rb:15:in `run!'
	from /opt/metalware/src/cli.rb:52:in `run!'
	from /opt/metalware/src/cli.rb:47:in `run'
	from bin/metal:28:in `<main>'
```

**Comment by bobwhitelock** [comment]
@WilliamMcCumstie can you take a look at this? Looks like an issue with validation for `choice` questions; afaics the answer should be valid.

Somewhat related, the fact that this was able to be saved but the validation fails shows why https://trello.com/c/eU4dOK4z/185-validate-answers-data-whenever-this-is-saved will be useful.

**Comment by bobwhitelock** [comment]
@WilliamMcCumstie: possibly the issue is that we don't handle `choice` values [here](https://github.com/alces-software/metalware/blob/c1d78a78026511965f1bc86f251b4c5358fc4fce/src/validation/answer.rb#L153-L162)?

**Comment by ColonelPanicks** [comment]
@WilliamMcCumstie I can confirm this is no longer an issue in develop branch (2017.1.0-rc1 tag). Happy to close this issue 😄

**Comment by WilliamMcCumstie** [comment]
Issued was fixed in #185
